### PR TITLE
commit page: migrate to use apollo

### PR DESCRIPTION
**Pull request by limitedmage**
Migrate the commit page to use Apollo `useQuery` instead of `requestGraphQL`

Still to do in future PR is to remove usage of `FilteredConnection` in favor of `useConnection`.

## Test plan

Verify manually and with integration tests.

## App preview:

- [Web](https://sg-web-jp-commitpageapollo.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.00% (-0.65 kb) | -0.01% (-0.65 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/jp/commitpageapollo/compare-report.html;tab=live_object) for a full comparison between the commits 04dcda4 and 4601469 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/commit/RepositoryCommitPage.tsx [hunk 1]
```diff
@@ -43,68 +42,27 @@ import {
...
-            map(({ data, errors }) => {
-                if (!data || !data.node) {
-                    throw createAggregateError(errors)
-                }
-                if (data.node.__typename !== 'Repository') {
-                    throw new Error(`Node is a ${data.node.__typename}, not a Repository`)
-                }
-                if (!data.node.commit) {
-                    // Filter out any revision not found errors, they usually come in multiples when searching for a commit, we want to replace all of them with 1 "Commit not found" error
-                    // TODO: Figuring why should we use `errors` here since it is `undefined` in this place
```
Comment (philipp-spiess) [inline client/web/src/repo/commit/RepositoryCommitPage.tsx:82]: lol nice cleanup

**Review by philipp-spiess — APPROVED** [review]
NASA LGTM

**Review by erzhtor — APPROVED** [review]
👍

**Pull request merged**